compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1163,6 +1163,9 @@ pub enum LocalInfo<'tcx> {
     /// (with no intervening statement context).
     // FIXME(matthewjasper) Don't store in this in `Body`
     BlockTailTemp(BlockTailInfo),
+    /// A temporary created during evaluating `if` predicate, possibly for pattern matching for `let`s,
+    /// and subject to Edition 2024 temporary lifetime rules
+    IfThen2024Temp { if_then: HirId },
```

Comment:
Remark [REV3,ED2024]: mentions of Edition 2024.

Reply:
Renamed